gpflow/likelihoods/base.py

```diff
@@ -366,15 +382,36 @@ class ScalarLikelihood(Likelihood):
 
     def __init__(self, **kwargs):
         super().__init__(latent_dim=None, observation_dim=None, **kwargs)
-        self.num_gauss_hermite_points = 20
 
     @property
-    def num_gauss_hermite_points(self):
+    def num_gauss_hermite_points(self) -> int:
+        warnings.warn(
+            "The num_gauss_hermite_points property is deprecated; access through the `quadrature` attribute instead",
+            DeprecationWarning,
+        )
+
+        if not isinstance(self.quadrature, NDiagGHQuadrature):
+            raise TypeError(
+                "Can only query num_gauss_hermite_points if quadrature is a NDiagGHQuadrature instance"
+            )
         return self.quadrature.n_gh
 
     @num_gauss_hermite_points.setter
-    def num_gauss_hermite_points(self, n_gh):
-        self.quadrature = NDiagGHQuadrature(1, n_gh)
+    def num_gauss_hermite_points(self, n_gh: int):
```

Comment:
Can we not simply delete this instead of introducing deprecated code. I don't think many people changed the number of gh points on the fly any way.

Reply:
Then it'd be a breaking change though, and worst of all one users wouldn't even notice (they'd assign a different number, but it simply wouldn't have an effect anymore). If it's reasonably easy to add a backwards-compatible shim like in this case, that gives a deprecation warning, I think we should.